packages/vscode-extension/src/debugging/DebugSession.ts

```diff
@@ -36,7 +36,13 @@ export class DebugSession implements Disposable {
   }
 
   public dispose() {
-    this.session && debug.stopDebugging(this.session);
+    let sessionExists = false;
+    try {
+      sessionExists = Boolean(this.session);
+    } catch {
+      // ignore errors as we are only interested if the session exists
+    }
+    sessionExists && debug.stopDebugging(this.session);
```

Comment:
we should just use `this.vscSession` here this is class's internal method, so there's no reason to use public interface and then workaround side effects